### PR TITLE
Client specification: point changelog to tags, add maintainer note

### PR DESCRIPTION
**Pull request by mebeim**
This converts the links in the changelog of the client spec to point to version tags instead of commit hashes as suggested by @bl-ue in #5428, and also includes a maintainer note in the form of a comment for future reference when updating the changelog. 

I've already pushed the tags from v1.0 to v1.5, which should be visible to everyone [here](https://github.com/tldr-pages/tldr/tags).

### CODE REVIEW

**Comment by bl-ue** [comment]
Also see my comment here: https://github.com/tldr-pages/tldr/pull/5428#issuecomment-803161951

File: CLIENT-SPECIFICATION.md
Comment (bl-ue) [inline CLIENT-SPECIFICATION.md:218]:
```suggestion
Keep the changelog links pointing to this document under the appropriate
```

**Comment by mebeim** [comment]
Quoting your comment from #5428 just to keep the discussion here:

> Hmm...I didn't get notified about this. @mebeim I think we need to create an actual release, not just a tag thinking
>
> I actually get GH notifications when I'm watching a repository and they create a release. For example, I was notified when https://github.com/kubernetes/kubernetes/releases/tag/v1.20.5 was released.

Makes sense. I should add this to the maintainer note in the comment then. Should we go ahead and just create a v1.5 release? Previous versions don't strictly need an actual release, but we could do that too.

**Comment by bl-ue** [comment]
Yes, I think we should. 👍🏻  No one needs to be notified about previous spec versions because the changes made in v1.0-1.4 are incorporated into v1.5.

**Comment by mebeim** [comment]
Done and done :+1:

**Review by bl-ue — APPROVED** [review]

**Review by navarroaxel — APPROVED** [review]

**Review by sbrl — APPROVED** [review]

**Comment by sbrl** [comment]
Is there anything else specific that needs doing after we merge this?

**Comment by bl-ue** [comment]
I don't think so, @sbrl. 👍🏻

**Pull request merged**
